lib/pry/cli.rb

```diff
@@ -59,11 +59,19 @@ def parse_options(args = ARGV)
           raise NoOptionsError, "No command line options defined! Use Pry::CLI.add_options to add command line options."
         end
 
-        self.input_args = args
+        index = args.index { |cli_arg| %w[- --].include?(cli_arg) }
+        if index
+          @pass_argv = true
+          slop_args = args[0...index]
+          self.input_args = args.replace(args[index + 1..-1])
+        else
+          @pass_argv = false
+          self.input_args = slop_args = args
```

Comment:
Why do we need to assign `slop_args`? It seems like on line 74 we can just use `self.input_args`.

Reply:
On the line you happened to comment on, we don't need `slop_args` but in the branch above this one we do.
`slop_args` is everything Slop should parse, and `self.input_args` is everything left over (ARGV) - afaik we have to separate them especially because Slop mutates its argument.

Reply:
I tried to remove `slop_args`, it doesn't work out.